Source/CarthageKit/Project.swift

```diff
@@ -531,6 +572,31 @@ public final class Project {
 			}
 	}
 
+	/// Emits the framework URL if it matches the local swift version and `empty` if not.
+	internal func matchingSwiftVersionURL(_ frameworkURL: URL) -> SignalProducer<URL, CarthageError> {
+		return SignalProducer.combineLatest(swiftVersion, frameworkSwiftVersion(frameworkURL))
+			.flatMap(.merge) { swiftVersion, frameworkVersion in
+				return swiftVersion == frameworkVersion
+					? SignalProducer(value: frameworkURL)
+					: SignalProducer(error: .incompatibleFrameworkSwiftVersions(local: swiftVersion, framework: frameworkVersion))
+			}
+			.take(first: 1)
+	}
+
+	/// Determines the Swift version of a framework at a given `URL`.
+	internal func frameworkSwiftVersion(_ frameworkURL: URL) -> SignalProducer<String, CarthageError> {
+		guard
+			let swiftHeaderURL = frameworkURL.swiftHeaderURL(),
+			let data = try? Data(contentsOf: swiftHeaderURL),
+			let contents = String(data: data, encoding: .utf8),
+			let swiftVersion = Project.parseSwiftVersionCommand(output: contents)
+		else {
+			return SignalProducer(error: .unknownFrameworkSwiftVersion)
+		}
+
+		return SignalProducer(value: swiftVersion)
+	}
```

Comment:
Do you think it makes sense for a `Project` to be able to determine it's framework's Swift version? Should I leave that in here or move it all over to `Xcode.swift`.